tests/components/accuweather/test_sensor.py

```diff
@@ -0,0 +1,598 @@
+"""Test sensor of AccuWeather integration."""
+from datetime import timedelta
+import json
+
+from homeassistant.components.accuweather.const import (
+    ATTRIBUTION,
+    CONCENTRATION_PARTS_PER_CUBIC_METER,
+    DOMAIN,
+    LENGTH_MILIMETERS,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_DEVICE_CLASS,
+    ATTR_ENTITY_ID,
+    ATTR_ICON,
+    ATTR_UNIT_OF_MEASUREMENT,
+    DEVICE_CLASS_TEMPERATURE,
+    LENGTH_METERS,
+    SPEED_KILOMETERS_PER_HOUR,
+    STATE_UNAVAILABLE,
+    TEMP_CELSIUS,
+    TIME_HOURS,
+    UNIT_PERCENTAGE,
+    UV_INDEX,
+)
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed, load_fixture
+from tests.components.accuweather import init_integration
+
+
+async def test_sensor_without_forecast(hass):
+    """Test states of the sensor without forecast."""
+    await init_integration(hass)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("sensor.home_cloud_ceiling")
+    assert state
+    assert state.state == "3200"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-fog"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == LENGTH_METERS
+
+    entry = registry.async_get("sensor.home_cloud_ceiling")
+    assert entry
+    assert entry.unique_id == "0123456-ceiling"
+
+    state = hass.states.get("sensor.home_precipitation")
+    assert state
+    assert state.state == "0.0"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == LENGTH_MILIMETERS
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-rainy"
+    assert state.attributes.get("type") is None
+
+    entry = registry.async_get("sensor.home_precipitation")
+    assert entry
+    assert entry.unique_id == "0123456-precipitation"
+
+    state = hass.states.get("sensor.home_pressure_tendency")
+    assert state
+    assert state.state == "falling"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:gauge"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == "accuweather__pressure_tendency"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)